daemon/daemon.go

```diff
@@ -819,6 +820,12 @@ func (d *Daemon) init() error {
 			return err
 		}
 
+		// Remove any old sockops and re-enable with _new_ programs
+		sockops.SockmapDisable()
```

Comment:
The *disable* logic will remove sockets from the sockmap and all messages will be processed through the normal L3/L4 stack e.g. via veth pairs. This ensures policy is still enforced. The *enable* logic (SockmapEnable, SkmsgEnable, ...) does not add existing sockets into the map. So the sockets that were removed will never get back into a sockmap.

This would be a nice feature request. https://github.com/cilium/cilium/issues/6092